.github/workflows/changelog.yml

```diff
@@ -0,0 +1,69 @@
+name: Generate changelog
+
+on:
+  workflow_dispatch:
+    inputs:
+
+      repository:
+        description: "Repository in which the changelog will be added."
+        required: false
+        default: 'aws'
+
+      ref:
+        description: |
+          Branch or ref to clone. This is where the changelog will be added and
+          ultimately committed.
+        required: true
+
+      type:
+        description: |
+          Type should be one of: feature, bugfix, enhancement, api-change
+          feature: A larger feature or change in behavior, usually resulting in a
+                   minor version bump.
+          bugfix: Fixing a bug in an existing code path.
+          enhancement: Small change to an underlying implementation detail.
+          api-change: Changes to a modeled API.
+        required: true
```

Comment:
It would be great if we could do the options here as these are a known set of values: https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#providing-inputs